File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/core/RendererSurface.kt
package dev.destack.runtime.android.core

/**
 * The primary renderer surface family for one Android host embedder.
 */
public enum class RendererSurfaceKind {
    /**
     * One `SurfaceView` backed host surface.
     */
    SurfaceView,

    /**
     * One `TextureView` backed host surface.
     */
    TextureView,

    /**
     * One `SurfaceControlViewHost` or similar embedded host surface.
     */
    EmbeddedHostSurface,
}

/**
 * The primary renderer surface for one Android host embedder.
 */
public data class RendererSurface(
    /**
     * The primary renderer surface kind.
     */
    val kind: RendererSurfaceKind,

    /**
     * The stable logical identifier for this host surface.
     */
    val identifier: String,
)

File: language/runtime/apple/core/Sources/RuntimeHostAppleCore/Core/RendererSurface.swift
import Foundation

/// The primary renderer surface family for one host embedder.
public enum RendererSurfaceKind: String, Sendable, Codable {
  /// One Core Animation backed surface.
  case metalLayer
  /// One Metal view backed surface.
  case metalView
  /// One platform view backed surface.
  case platformView
}

/// The primary renderer surface for one host embedder.
public struct RendererSurface: Sendable, Hashable, Codable {
  /// The primary renderer surface kind.
  public let kind: RendererSurfaceKind
  /// The stable logical identifier for this host surface.
  public let identifier: String

  /// Create one renderer surface.
  public init(
    kind: RendererSurfaceKind,
    identifier: String
  ) {
    self.kind = kind
    self.identifier = identifier
  }
}
